main.py

```diff
@@ -474,7 +488,14 @@ def get_link_metadata(self, link_id):
     @rest('v3/links/<link_id>/metadata', methods=['POST'])
     def add_link_metadata(self, link_id):
         """Add metadata to a link."""
-        metadata = request.get_json()
+        try:
+            metadata = request.get_json()
+        except BadRequest:
+            return jsonify('The request body is not a well-formed JSON.'), 400
+        if metadata is None:
+            return jsonify('The request body mimetype is not'
+                           ' application/json.'), 400
```

Comment:
Does this really need to be repeated 3 times? Can it be refactored into a function?

Reply:
This check has been refactored into a function.